### PR TITLE
MULE-16364: ManagedStreams are not semantically scoped

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: integration/src/test/java/org/mule/test/streaming/AutoCloseCursorProviderTestCase.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import java.io.File;
+
+import javax.inject.Inject;
+
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class AutoCloseCursorProviderTestCase extends AbstractIntegrationTestCase {
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/streaming/AutoCloseCursorProviderTestCase.java]: allure

File: integration/src/test/java/org/mule/test/streaming/AutoCloseCursorProviderTestCase.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+  @Test
+  public void openManyStreamsInForeachAndDiscard() throws Exception {
+    String content = randomAlphanumeric(1024 * 1024);
+    File file = new File(temporaryFolder.getRoot(), "file.txt");
+    writeStringToFile(file, content);
+
+    flowRunner("openManyStreamsInForeachAndDiscard").run();
+
+    check(5000, 100, () -> {
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/streaming/AutoCloseCursorProviderTestCase.java]: 5000 and 100 are already the default values i think

Comment (marianogonzalez) [reply integration/src/test/java/org/mule/test/streaming/AutoCloseCursorProviderTestCase.java]: it's not. Plus I'm using the static version of it

Comment (elrodro83) [reply integration/src/test/java/org/mule/test/streaming/AutoCloseCursorProviderTestCase.java]: use `PollingProber.probe`

File: integration/src/test/java/org/mule/test/streaming/AutoCloseCursorProviderTestCase.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+  private static final int OPEN_PROVIDERS = 100;
+  private static StreamingStatistics statistics;
+
+
+  public static class AssertStatisticsProcessor implements Processor {
+
+    @Override
+    public CoreEvent process(CoreEvent event) throws MuleException {
+      check(10000, 100, () -> {
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/streaming/AutoCloseCursorProviderTestCase.java]: use `PollingProber.probe`

**Pull request merged**